src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+<?php
+namespace App\Api\Test;
+
+use ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Client as TestClient;
+use App\Api\Test\BaseApiTestCase;
+use App\Entity\Client;
+use App\Entity\Script;
+
+class ClientTest extends BaseApiTestCase
+{
+    public function testCreateClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testCreateClientAllParameters(): void 
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $scriptId = $this->getScriptId($httpClient, 'script_all_true');
```

Comment:
Just curious.

Why do you need to reference the script by name ? I suppose you created the script as a fixture or something like that.  Isn't there a way to directly request the ids of the created fixtures by using the fixture library or something like that?

Reply:
That's it, is a script created as fixture, but I have not found any other way to get the fixture's id

Reply:
I see, that's one of the caveats of auto generated Ids, they limit testability: https://github.com/nelmio/alice/issues/999

Ok, let's do something not very nice but a bit better.

As I can see in the authenticate method the httpclient can be created statically. So you could just put the logic to obtain the scripts inside ClientMother.

That way when you call something like ClientMother::withPreScripts() you can internally do whatever call you need to obtain the ids and assign them to the client's JSON.

Not very nice but a at least our ClientTest will be free of this noise.

Reply:
In that case the issue would be that I cannot get the iri by the name from the clientMother, it is a ApiTestCase's method ($this->findIriBy(Script::class, ['name' => $scriptName])), so I cannot think about a way to get the id of the script

Reply:
Ok, let's not invest more time with this. Keep on doing it as you are doing, we will figure out how to really solve this in a future.